src/mmw/js/src/draw/views.js

```diff
@@ -456,8 +456,9 @@ var WatershedDelineationView = Marionette.ItemView.extend({
                 displayAlert(message, modalModels.AlertTypes.warn);
             })
             .always(function() {
-                if (rwdClickedPoint) {
-                    map.removeLayer(rwdClickedPoint);
+                if (self.model.has('rwd-original-point')) {
```

Comment:
Since this process is repeated and requires multiple steps to be successful, consider putting it inside of a model instance method, so that you might be able to say `model.clearRwdMarker()` or similar, here and below.  If the model doesn't have a reference to `App`, it could accept a `map` argument for clearing the layer.